packages/@sanity/cli/src/cli.ts

```diff
@@ -97,6 +98,7 @@ export async function runCli(cliRoot: string, {cliVersion}: {cliVersion: string}
     cliRoot: cliRoot,
     workDir: workDir,
     corePath: await getCoreModulePath(workDir, cliConfig),
+    projectRootPath: cliConfig ? workDir : await getProjectRoot(),
```

Comment:
a potential (although probably quite small) perf hit by doing these in sequence, so you could consider moving them up and do something like:
```ts
const [corePath, projectRootPath] = Promise.all([
  getCoreModulePath(workDir, cliConfig),
  getProjectRoot()
])
```

Reply:
Aha! Good catch. I'll see if this can be easily refactored.